### PR TITLE
codetime metrics and event implementation

### PR DESCRIPTION
**Pull request by xavluiz**

### CODE REVIEW

File: SoftwareCo/SoftwareCo/Managers/DocEventManager.cs [hunk 1]
```diff
@@ -52,112 +50,146 @@ private async void InitPluginDataIfNotExists()
...
+            }
+
+            if (doc == null)
+            {
+                doc = (Selection.DTE != null && Selection.DTE.ActiveDocument != null) ? Selection.DTE.ActiveDocument : await PackageManager.GetActiveDocument();
+            }
+
+            if (doc == null)
+            {
+                return;
```
Comment (bjacobson26) [inline SoftwareCo/SoftwareCo/Managers/DocEventManager.cs:132]: Should this line move to 128?

File: SoftwareCo/SoftwareCo/Utils/Constants.cs [hunk 2]
```diff
@@ -5,8 +5,8 @@ internal static class Constants
         internal const string EditorName = "visualstudio";
         internal const long DEFAULT_SESSION_THRESHOLD_SECONDS = 60 * 15;
 
-        internal const string api_endpoint = "https://api.software.com";
-        internal const string url_endpoint = "https://app.software.com";
+        internal const string api_endpoint = "https://stagingapi.software.com"; // "https://api.software.com";
```
Comment (bjacobson26) [inline SoftwareCo/SoftwareCo/Utils/Constants.cs:8]: Remember to change this back before merging

File: SoftwareCo/SoftwareCo/Managers/DocEventManager.cs (hunk 1 above)
Comment (xavluiz) [reply SoftwareCo/SoftwareCo/Managers/DocEventManager.cs:132]: i'll add comments but there's a method called "OnChangeAsync" that should set the local "doc" when tabs change, but if its not set yet by the time it hits one of these other listener handlers we'll try to set it. if its still not set we'll fall out of the method. its a bit of space and conditions but the alternative is exceptions that can crash the IDE

File: SoftwareCo/SoftwareCo/Utils/Constants.cs (hunk 2 above)
Comment (xavluiz) [reply SoftwareCo/SoftwareCo/Utils/Constants.cs]: right, thanks

**Pull request merged**